gh-119260: Clarify is_dataclass Behavior for Subclasses in Documentation and Tests (#119480)
python/cpython: Doc/library/dataclasses.rst

.. function:: is_dataclass(obj)

>    Return ``True`` if its parameter is a dataclass (including subclasses of a dataclass)
>    or an instance of one, otherwise return ``False``.

Reviewer: ```suggestion
   Return ``True`` if its parameter is a dataclass (including subclasses of a
   dataclass) or an instance of one, otherwise return ``False``.
```

.. function:: is_dataclass(obj)

>    Return ``True`` if its parameter is a dataclass, an instance of a dataclass,
>    or a subclass of a dataclass, otherwise return ``False``. Note that subclasses
>    of dataclasses inherit the properties of the dataclass so will return ``True``.

Reviewer: This seems more verbose than necessary, but also not fully clear (that parameter can be an instance of a subclass of a dataclass too.) Suggested change:
```suggestion
   Return ``True`` if its parameter is a dataclass (including subclasses of a dataclass)
   or an instance of one, otherwise return ``False``.
```

Author: Modified the comment as per your suggestion !
